main/src/ca/uwaterloo/flix/util/collection/ListMap.scala

```diff
@@ -120,6 +120,13 @@ case class ListMap[K, V](m: Map[K, List[V]]) {
     } yield (k, v)
   }
 
+  /**
+    * Returns a new ListMap with the key-value pairs `k -> v` if `p(k, v)` holds.
+    */
```

Comment:
Does this preserve the order of the lists?

Reply:
yes

Reply:
there are tests